zerver/lib/bugdown/__init__.py

```diff
@@ -891,11 +905,7 @@ def run(self, root: Element) -> None:
             if (arguments.current_message is None
                     or not url_embed_preview_enabled_for_realm(arguments.current_message)):
```

Comment:
We are doing some checks here. So you cannot move `arguments.current_message.links_for_preview.add(url)` to top. Keep this section as it is.